nixos/modules/services/web-apps/stirling-pdf.nix

```diff
@@ -0,0 +1,168 @@
+{
+  config,
+  lib,
+  pkgs,
+  ...
+}:
+
+let
+  inherit (lib)
+    getExe
+    literalExpression
+    mkEnableOption
+    mkIf
+    mkMerge
+    mkOption
+    mkPackageOption
+    optional
+    ;
+
+  inherit (lib.types)
+    attrsOf
+    nullOr
+    path
+    port
+    str
+    submodule
+    ;
+
+  cfg = config.services.stirling-pdf;
+in
+
+{
+  meta.maintainers = with lib.maintainers; [ thubrecht ];
+
+  options.services.stirling-pdf = {
+    enable = mkEnableOption "Stirling-PDF, a web application for manipulating PDF files";
+
+    package = mkPackageOption pkgs "stirling-pdf" { };
+
+    domain = mkOption {
+      type = str;
+      description = ''
+        The domain for the application.
+      '';
+    };
+
+    port = mkOption {
+      type = port;
+      default = 8080;
+      description = ''
+        Port that Stirling-PDF will listen on.
+      '';
+    };
+
+    environment = mkOption {
+      type = attrsOf str;
```

Comment:
this type seems too restrictive; since the `environment` option is used only to set `systemd.service.stirling-pdf.environment`, [which also allows `null` and paths](https://search.nixos.org/options?channel=23.11&show=systemd.services.%3Cname%3E.environment&from=0&size=50&sort=relevance&type=packages&query=systemd.services.%3Cname%3E.environment).